File: mobile/agent-guidance.mjs
import {readFileSync} from 'node:fs';
export const guide=readFileSync(new URL('../src/issues/web/agent-guide.md',import.meta.url),'utf8');
export const script=readFileSync(new URL('../src/issues/web/agent-guide.js',import.meta.url),'utf8');
const head=readFileSync(new URL('../src/issues/web/agent-guide-head.html',import.meta.url),'utf8');
export function addAgentGuidance(html){
 const escaped=guide.replaceAll('&','&amp;').replaceAll('<','&lt;').replaceAll('>','&gt;');
 return html.replace('</head>',`${head}</head>`)
  .replace('</body>',`<pre hidden id="hey-boss-agent-guide">${escaped}</pre>\n</body>`);
}
export function agentGuidancePlugin(){
 return {name:'hey-boss-agent-guidance',transformIndexHtml:addAgentGuidance,
  generateBundle(){
   this.emitFile({type:'asset',fileName:'llms.txt',source:guide});
   this.emitFile({type:'asset',fileName:'agent-guide.js',source:script});
  }};
}

File: src/agent_guidance.rs
//! A discovery guide, not a second resource renderer. Keep it shared with mobile.
pub const GUIDE: &str = include_str!("issues/web/agent-guide.md");
pub const SCRIPT: &str = include_str!("issues/web/agent-guide.js");
const HEAD: &str = include_str!("issues/web/agent-guide-head.html");

pub fn decorate(html: &str) -> String {
    let guide = GUIDE
        .replace('&', "&amp;")
        .replace('<', "&lt;")
        .replace('>', "&gt;");
    html.replacen("</head>", &format!("{HEAD}</head>"), 1)
        .replacen(
            "</body>",
            &format!("<pre hidden id=\"hey-boss-agent-guide\">{guide}</pre>\n</body>"),
            1,
        )
}
